zeppelin-web/src/app/notebook/paragraph/paragraph.controller.js

```diff
@@ -830,6 +830,29 @@ function ParagraphCtrl ($scope, $rootScope, $route, $window, $routeParams, $loca
       // autocomplete on 'ctrl+.'
       $scope.editor.commands.bindKey('ctrl-.', 'startAutocomplete')
 
+      // Show autocomplete on tab
+      $scope.editor.commands.addCommand({
+        name: 'tabAutocomplete',
+        bindKey: {
+          win: 'tab',
+          mac: 'tab',
+          sender: 'editor|cli'
+        },
+        exec: function(env, args, request) {
+          let iColumnPosition = $scope.editor.selection.getCursor().column
+
+          // If user has pressed tab on first line char or if editor mode is %md, keep existing behavior
+          // If user has pressed tab anywhere in between and editor mode is not %md, show autocomplete
+          if (iColumnPosition && $scope.paragraph.config.editorMode !== 'ace/mode/markdown') {
```

Comment:
Some interpreters (language) may make sense bring autocompletion list anytime user press 'tab'.
Some interpreters may make sense bring autocompletion list user press 'tab' only after the dot (.).
Some interpreters may make sense bring autocompletion list list anytime user press dot (.).
Some interpreters does not make sense bring autocompletion on 'tab' key.
...
...

I'm not sure 'tab' is suitable completion key for all interpreters available, especially when we think about 3rd party interpreters that user created.

Are we sure that 'tab' is suitable completion key for all interpreters available in Zeppelin?